### PR TITLE
fix(atomic): make resultContext function generic

### PR DESCRIPTION
**Pull request by liviarett**
https://coveord.atlassian.net/browse/KIT-1551

### CODE REVIEW

File: packages/atomic/src/components/result-template-components/result-template-decorators.tsx [hunk 1]
```diff
@@ -81,11 +83,11 @@ const resultContextEventName = 'atomic/resolveResult';
  * @param element The element that the event is dispatched to, which must be the child of a rendered "atomic-result".
  * @returns A promise that resolves on initialization of the parent "atomic-result" element, or rejects when there is no parent "atomic-result" element.
  */
-export const resultContext = (element: Element) =>
-  new Promise<FoldedResult | Result>((resolve, reject) => {
-    const event = buildCustomEvent<ResultContextEventHandler>(
+export function resultContext<T = Result>(element: Element) {
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/result-template-components/result-template-decorators.tsx:86]: We should constrain it
```suggestion
export function resultContext<T extends Result | FoldedResult = Result>(element: Element) {
```

**Review by ThibodeauJF — APPROVED** [review]
Thanks!

File: packages/atomic/src/components/result-template-components/result-template-decorators.tsx (hunk 1 above)
Comment (liviarett) [reply packages/atomic/src/components/result-template-components/result-template-decorators.tsx:86]: woaaah nice

**Review by olamothe — APPROVED** [review]

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/result-template-components/result-template-decorators.tsx [hunk 2]
```diff
@@ -81,11 +83,11 @@ const resultContextEventName = 'atomic/resolveResult';
  * @param element The element that the event is dispatched to, which must be the child of a rendered "atomic-result".
  * @returns A promise that resolves on initialization of the parent "atomic-result" element, or rejects when there is no parent "atomic-result" element.
  */
-export const resultContext = (element: Element) =>
-  new Promise<FoldedResult | Result>((resolve, reject) => {
-    const event = buildCustomEvent<ResultContextEventHandler>(
+export function resultContext<T extends Result | FoldedResult = Result>(element: Element) {
```
[the remark this answers is not in the document]
Comment (liviarett) [reply packages/atomic/src/components/result-template-components/result-template-decorators.tsx:86]: not sure what's going on, and I hadn't even noticed. trust me, I haven't been linting my code manually this whole time hahaha the formatOnSave is on, but it's not even flagging it locally!

[the remark this answers is not in the document]
Comment (liviarett) [reply packages/atomic/src/components/result-template-components/result-template-decorators.tsx:86]: aaah! it was committed straight on github, that's why! and I never pulled locally. I wonder if we could add the lint fixing command on github commits?

**Pull request merged**